### DISCUSSION TITLE
Face-api.js In React native

### DISCUSSION THREAD

**Discussion by JayasimhaMH** [vladmandic/face-api#164] [Q&A]
Hi , Vladmandic , Can Faceapi.js supports react native application ? If it is supported please help me in that . Here am trying to implement the face match process in react native application . Please kindly help me .

**Comment by vladmandic** [comment]
until recently, `tfjs-react-native` was stuck using tfjs v3 and it was not compatible  
but `tfjs-react-native` was recently updated to use tfjs v4  

so in theory, it should work  
but i have not tested it in no way nor i have the resources to experiment with react-native at the moment

**Reply by JayasimhaMH** [reply]
I'm Try to this way but Face-api.js Modules Not Loading, I got this  errors ( open failed :ENOENT ( No such file or directory), ) and  BufferToImages -expected Buf to be of type :Blob)

**Comment by JayasimhaMH** [comment]
const Facematch = async () => {
        try {
          
          let filePathRoot = RNFS.DocumentDirectoryPath + '../../assets/model';

          faceapi.nets.ssdMobilenetv1.loadFromDisk(filePathRoot),
          faceapi.nets.faceLandmark68Net.loadFromDisk(filePathRoot),
          faceapi.nets.faceRecognitionNet.loadFromDisk(filePathRoot)


           const imagePath1 = '../../assets/person-1.jpg';
           const imagePath2 = '../../assets/person-2.jpg';

           const iamges1 = await faceapi.bufferToImage(imagePath1)
           const iamges2 = await faceapi.bufferToImage(imagePath2

           const image1 = await canvas.loadImage(canvas1);
           const image2 = await canvas.loadImage(canvas2);

           const face1 = await faceapi.detectSingleFace(image1).withFaceLandmarks().withFaceDescriptor();
           const face2 = await faceapi.detectSingleFace(image2).withFaceLandmarks().withFaceDescriptor();

  
        const result = await compareFaces(imagePath1, imagePath2);

        if (!face1 || !face2) {
        return 'Face not detected in one or both images.';
        }

        const distance = faceapi.euclideanDistance(face1.descriptor, face2.descriptor);
        const threshold = 0.6;
        faceapi.tf.enableProdMode() 
        faceapi.tf.ENV.set('DEBUG', false);
        compareFaces().catch(err => console.error(err));
        performFaceMatchingface().catch(err => console.error(err));
       
        if (distance <= threshold) {
          return 'Same person detected.';
        } else {
          return 'Different person detected.';
        }
        
      } catch (error) {
        console.error('Error occurred during face matching:', error);
       }
       };

**Reply by timotismjntk** [reply]
hey bro @JayasimhaMH i manage to build for my self face recognition in react native and maybe you can email: me to discuss timoticrew@gmail.com
